Added troubleshooting section for lock issue on network fs (#1842)
iterative/dvc.org: content/docs/user-guide/troubleshooting.md

[cache directory](/doc/command-reference/config#cache) doesn't exist. It is
usually created automatically by DVC commands that need it, but you can create
it manually (e.g. `mkdir .dvc/cache`) to enable this check.

> ## Unable to acquire lock {#lock-issue}

Reviewer: @harsh8398 looks great!!

The only thing that comes to my mind to change a bit is to expand on `If that is not the case, it usually means that DVC was terminated abruptly and manually ...` WDYT?

Otherwise looks great.

[cache directory](/doc/command-reference/config#cache) doesn't exist. It is
usually created automatically by DVC commands that need it, but you can create
it manually (e.g. `mkdir .dvc/cache`) to enable this check.

## Unable to acquire lock {#lock-issue}

> You may encounter error message saying `Unable to acquire lock` if you have

Reviewer 2: ```suggestion
You may encounter an error message saying `Unable to acquire lock` if you have
```

another DVC process running in the project. If that is not the case, it usually

> means that DVC was terminated abruptly and manually removing lockfile
> `.dvc/tmp/lock` should resolve the issue. If the issue still persists then it

Reviewer 2: ```suggestion
means that DVC was terminated abruptly and manually removing the lock file in
`.dvc/tmp/lock` should resolve the issue. If the issue still persists then it
```

> `.dvc/tmp/lock` should resolve the issue. If the issue still persists then it
> may be the case that you are running DVC on some network filesystem like NFS,
> Lustre, etc. If that is the case, the solution is to enable `core.hardlink_lock`
> which can be done by running following command:

Reviewer 2: ```suggestion
`.dvc/tmp/lock` should resolve the issue.

If the issue still persists then it may be the case that you are running DVC on
some network filesystem like NFS, Lustre, etc. If so, the solution is to enable
`core.hardlink_lock` which can be done by running following command:
```

Reviewer 2: How about separating the paragraphs ^ @shcheklein ?

Reviewer: I don't have any opinion on this @jorgeorpinel. Up to you, guys :)

[cache directory](/doc/command-reference/config#cache) doesn't exist. It is
usually created automatically by DVC commands that need it, but you can create
it manually (e.g. `mkdir .dvc/cache`) to enable this check.

> ## Lock issue on network file system {#lock-issue}

Reviewer: Let's make it general (and we might be able to simplify the ERROR message in DVC itself). I would mention here that in most cases it is still about `.dvc/tmp/lock` file and about another DVC process running. After that mention about filesystems like Lustre.

The first paragraph we should start with a brief mention of those two.

Author: Are you suggesting we can just put lock error message in dvc as "Unable to acquire lock. Aborted command execution. For troubleshooting, please visit - <link>" and make troubleshooting entry more general for possible fixes?

Reviewer: @harsh8398 yes, exactly!. I'm still not sure about the `Aborted command execution` - if we should say something like ... `Most likely another DVC process is running or was terminated abruptly. Check the page ....` ...

Author: @shcheklein I have updated the doc. I'll add the link in the error message in DVC itself once I'll be able to resolve the test case issue. Thanks.

> ## Lock issue when running DVC on network filesystem {#lock-issue}

Reviewer 2: ```suggestion
## Lock issue on network file system {#lock-issue}
```

> Users may encounter following lock error when running DVC on some network
> filesystem (NFS, Lustre, etc.)

Reviewer 2: ```suggestion
Users may encounter the following lockfile error when running DVC on some
network filesystem (NFS, Lustre, etc.)
```

```
ERROR: cannot perform the command because another DVC process seems to be
running on this project. If that is not the case, manually remove
.dvc/tmp/lock and try again.
```

The solution is to enable `core.hardlink_lock` which can be done by running

> following command:

Reviewer 2: Missing the remark about this persisting even after removing .dvc/tmp/lock
